### ISSUE TITLE
<bit> for simd (popcount, bit_ceil, ...)

### ISSUE THREAD

**Issue by yaozhongxiao** [mattkretz/std-simd-feedback#19]
The bits of the values of the primary types such as int， long,...  can be evaluated by popcount-like operations.
in this way, it will be useful to support bit-wise operations on simd beside simd_mask with popcount.

**Comment by mattkretz** [comment]
I agree. Basically all of [`<bit>`](https://en.cppreference.com/w/cpp/header/bit) should be overloaded on `simd<T>` where `T` is constrained as currently specified.

**Comment by danieltowner** [comment]
I'm will write this up as a paper. I assume it should duplicate the ideas of the `<cmath>` section of the main std::simd paper, but be tweaked to reference <bit> overloads instead, and that the paper should be marked for the attention of LEWG?

**Comment by mattkretz** [comment]
I don't fully understand. You will have to add overloads to all the functions in `<bit>`. Where these overloads go (header and section in the standard) :shrug: no idea.
And yes, LEWG.

**Comment by danieltowner** [comment]
Edited my original comment - some markup made the text disappear.

Your simd paper has a section `[simd.math]` which lists all the changes needed to support that header with simd. I assume that I should propose wording which introduces another section called `[simd.bit]` which does something similar for those overloads?

**Comment by mattkretz** [comment]
I don't think LWG would appreciate wording like that. `<cmath>` is special because of the way C++ defers to the C specification. We don't need that here. So just define new functions (or rather overloads) like you specify `permute` etc.

**Comment by danieltowner** [comment]
https://isocpp.org/files/papers/P2933R1.html
